### PR TITLE
Order missing fields first in a desc sort (#2174)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make update_requirements` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit

Primary reviewer (after pushing merge commit)

- [ ] Moved linked issue to Merged
- [ ] Pushed merge commit to Github
- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Review by amarjandu — APPROVED** [review]

File: test/service/test_response.py [hunk 1]
```diff
@@ -1414,6 +1414,46 @@ def test_ordering(self):
...
+
+        def extract_cell_line_types(response_json):
+            # For each hit yield the 'cellLineType' value or None if not present
+            for hit in response_json['hits']:
+                if hit['cellLines']:
+                    yield one(hit['cellLines'])['cellLineType']
+                else:
+                    yield None
+
+        def _test(order, expected_values):
```
Comment (hannes-ucsc) [inline test/service/test_response.py:1441]: same as in other PR.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Has conflicts. No bump.

File: test/service/test_response.py [hunk 2]
```diff
@@ -1417,6 +1417,44 @@ def test_ordering(self):
...
+        results with the hits missing the field placed at the end of a
+        ascending sort and the beginning of a descending sort.
+        """
+        ascending_values = [
+            ['induced pluripotent'],
+            ['induced pluripotent'],
+            ['primary', 'stem cell-derived'],
+            None,  # The last 4 hits don't have any 'cellLines' inner entities
+            None,  # so for purposes of this test we use None to represent
+            None,  # that there is a hit however it has no 'cellLineType'.
```
Comment (hannes-ucsc) [inline test/service/test_response.py:1432]:
```suggestion
            None,  # that there is a hit but that it has no 'cellLineType'.
```

Unless your form is grammatically correct. Not sure though.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo attempt to reproduce #2174.

**Pull request merged**